### PR TITLE
Fixing vttest for go / vitess.

### PR DESCRIPTION
**Pull request by alainjobart**
@sougou 
Fixing a few things:
- moving import of gRPC client to environment.go, so we can do other protocols in other places.
- topoFlags are the same everywhere now, moving them out of environment.go
- adding host so local connections work without socket.

### CODE REVIEW

**Comment by alainjobart** [comment]
Gonna merge this as soon as travis passes, as it's needed in google3 import.

File: py/vttest/mysql_db_mysqlctl.py [hunk 1]
```diff
@@ -71,3 +71,10 @@ def port(self):
 
   def unix_socket(self):
     return os.path.join(self._directory, 'vt_0000000001', 'mysql.sock')
+
+  def config(self):
+    return {
+        'username': self.username(),
+        'password': self.password(),
+        'socket': self.unix_socket(),
+    }
```
Comment (sougou) [inline py/vttest/mysql_db_mysqlctl.py]: Heh. I had this on my todo list, and forgot.

File: py/vttest/mysql_db.py [hunk 2]
```diff
@@ -35,3 +35,7 @@ def port(self):
 
   def unix_socket(self):
     raise NotImplementedError('MySqlDB is the base class.')
+
+  def config(self):
+    """Returns the json config to output."""
+    raise NotImplementedError('MySqlDB is the base class.')
```
Comment (sougou) [inline py/vttest/mysql_db.py]: I was going to delete this class. I think this kind of stuff is not necessary in Python, but wanted to check with you.

**Comment by sougou** [comment]
lgtm

File: py/vttest/mysql_db.py (hunk 2 above)
Comment (alainjobart) [inline py/vttest/mysql_db.py]: It's not necessary, but it's a good place to document the interface, and if people forget to implement the method, they get a nice message and pointer to the doc too. So I'd rather keep it that way. :)

**Pull request merged**